src/Build/BackEnd/Node/RarNode.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO.Pipes;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.Build.BackEnd;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Internal;
+using Microsoft.Build.Shared;
+using Microsoft.Build.Tasks.ResolveAssemblyReferences.Server;
+
+namespace Microsoft.Build.Execution
+{
+    public sealed class RarNode : INode
+    {
+        /// <summary>
+        /// The amount of time to wait for the client to connect to the host.
+        /// </summary>
+        private const int ClientConnectTimeout = 60000;
+
+        public NodeEngineShutdownReason Run(bool nodeReuse, bool lowPriority, out Exception shutdownException)
+        {
+            shutdownException = null;
+            using CancellationTokenSource cts = new CancellationTokenSource();
+            string pipeName = CommunicationsUtilities.GetRarPipeName(nodeReuse, lowPriority);
+            RarController controller = new RarController(pipeName, NamedPipeUtil.CreateNamedPipeServer);
+
+            Task<int> rarTask = controller.StartAsync(cts.Token);
+
+            Handshake handshake = NodeProviderOutOfProc.GetHandshake(enableNodeReuse: nodeReuse,
+                                                                     enableLowPriority: lowPriority, specialNode: true);
+            Task<NodeEngineShutdownReason> msBuildShutdown = RunShutdownCheckAsync(handshake, cts.Token);
+
+            int index = Task.WaitAny(msBuildShutdown, rarTask);
+            cts.Cancel();
```

Comment:
Comment this, please?

Reply:
The though here was to cancel all tasks so they don't stay in background, but it is probably somehow useless, because when we get to this point, the process is doomed to die...